### PR TITLE
Make vtgate streamlog buffer configurable

### PR DESCRIPTION
**Pull request by DeathBorn**
If vtgate streamlog channel becomes full, query logs will be dropped. 
This makes its size configurable.

### CODE REVIEW

**Comment by deepthi** [comment]
@DeathBorn this also needs the DCO to be fixed.
The easiest way to avoid this on future PRs is to do your git commits with `-s` or `--signoff`

**Comment by DeathBorn** [comment]
@deepthi pr fixed, sorry for force push, but this is just one commit.

**Comment by systay** [comment]
seems like unit tests are failing, expecting the `QueryLogger` to already have been initialised. care to look at some of those?

**Comment by DeathBorn** [comment]
seems `vtexplain` is not calling `initQueryLogger`

**Comment by DeathBorn** [comment]
@systay ping :)

**Review by systay — APPROVED** [review]

File: go/vt/vtgate/querylog.go [hunk 1]
```diff
@@ -34,13 +34,17 @@ var (
...
 
 	// QueryLogger enables streaming logging of queries
-	QueryLogger = streamlog.New("VTGate", 10)
+	QueryLogger *streamlog.StreamLogger
 
 	// queryLogToFile controls whether query logs are sent to a file
 	queryLogToFile = flag.String("log_queries_to_file", "", "Enable query logging to the specified file")
+
+	// queryLogBufferSize controls how many query logs will be buffered before dropping them if logging is not fast enough
+	queryLogBufferSize = flag.Int("querylog-buffer-size", 10, "Maximum number of buffered query logs before throttling log output")
```
Comment (harshit-gangal) [inline go/vt/vtgate/querylog.go:43]: @deepthi @ajm188 are we using `dash` or `underscore` for new flags?

**Review by harshit-gangal — APPROVED** [review]

**Comment by harshit-gangal** [comment]
@DeathBorn you need to fix a e2e test regarding flag

```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -259,2 +259,4 @@
        	            	 	how often try to remove old logs (default 1h0m0s)
        	            	+  --querylog-buffer-size int
        	            	+	Maximum number of buffered query logs before throttling log output (default 10)
        	            	   --querylog-filter-tag string
        	Test:       	TestHelpOutput
```

File: go/vt/vtgate/querylog.go (hunk 1 above)
Comment (harshit-gangal) [reply go/vt/vtgate/querylog.go:43]: `dash` :+1:

**Pull request merged**